url64b10c0106_schlabberdog_blocks_tgz-pJ8-com_github_users_schlabberdog_blocks_ui_CliUIJ8/src/com/github/users/schlabberdog/blocks/ui/IJGUI.java

```diff
@@ -0,0 +1,284 @@
+		//zuerst den button wieder deaktivieren
+		nextButton.setEnabled(false);
+		pathStopLength.setEnabled(false);
+
+		if(isSolverRunning) {
+			//ggf. veränderten wert erneut kopieren
+			stopAtCount = ((Number)pathStopLength.getValue()).intValue();
+			//die uhr weiterlaufen lassen
+			stopwatch.start();
+			//dann den solver auch weiter laufen lassen
+			solverThread.interrupt();
+		}
+    }
+
+    public synchronized void startSolve() {
+        //gui deaktivieren
+        fastForwardButton.setEnabled(false);
+        stackLimiterSpinner.setEnabled(false);
+        avoidWorseCheckbox.setEnabled(false);
+		pathStopLength.setEnabled(false);
+		//uhr zurücksetzen
+		stopwatch.reset();
+        //werte kopieren
+        solver.setStackDepthLimit(((Number) stackLimiterSpinner.getValue()).intValue());
+        solver.setAvoidWorseStacks(avoidWorseCheckbox.isSelected());
+		stopAtCount = ((Number)pathStopLength.getValue()).intValue();
+        //für den solver benutzen wir einen eigenen Thread
+		solverThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+				solver.solve();
+            }
+        });
+		solverThread.setUncaughtExceptionHandler(this);
+		solverThread.start();
+
+    }
+
+    public synchronized void doStep() {
+
+    }
+
+    private void createUIComponents() {
+        boardView = new BoardView(board);
+    }
+
+    public static IJGUI Create(Board board, Solver solver) {
+        IJGUI gui = new IJGUI(board,solver);
+
+        solver.setDelegate(gui);
+
+        JFrame frame = new JFrame("IJGUI");
+        frame.setContentPane(gui.root);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.pack();
+		frame.setMinimumSize(frame.getSize());
+        frame.setVisible(true);
+
+        gui.frame = frame;
+
+        return gui;
+    }
+
+	@Override
+	public synchronized void solverStarted(Solver solver) {
+		timer.start();
+		stopwatch.start();
+		isSolverRunning = true;
+	}
+
+	@Override
+	public void solutionImproved(Solver solver, int solSize) {
+		//System.out.println("Better solution: "+solSize);
+		bestSolution = solver.getStepList();
+
+		//jetzt können wir den solver trappen, wenn er halten soll
+		if(stopAtCount > 0 && solver.getBestPathLength() <= stopAtCount) {
+			//zuerst mal die uhr anhalten
+			stopwatch.stop();
+			//lösung im GUI thread anzeigen
+			SwingUtilities.invokeLater(new Runnable() {
+				@Override
+				public void run() {
+					//anzeige der lösung starten
+					IJSolutionBrowser.Create(board.copy(),initialState,bestSolution);
+					//weiter button aktivieren
+					nextButton.setEnabled(true);
+					//verändern des halte-limits erlauben
+					pathStopLength.setEnabled(true);
+					//gui neu zeichnen
+					validateButtons();
+				}
+			});
+			//solver anhalten
+			try {
+				//noinspection InfiniteLoopStatement
+				while(true) {
+					Thread.sleep(1000); //jaja, wait() ist vieeeel cooler...
+				}
+			} catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)